src/pages/support-commands/details/index.tsx

```diff
@@ -0,0 +1,217 @@
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import classNames from "classnames";
+import { useAppDispatch, useAppSelector } from "store/store";
+import { formatToTitleCase, getStatusLabel } from "utils/utils";
+import { Box, Button, CircularProgress, Grid, Snackbar, Alert } from "@mui/material";
+import { Card } from "../components/Card";
+import dayjs from "dayjs";
+import TheaterComedyIcon from "@mui/icons-material/TheaterComedy";
+
+import { PROFILE_DETAILS_KEYS } from "../config";
+
+import UserDetailsModal from "../components/UserDetailsModal";
+import {
+  clearSuccess,
+  fetchProfileDetails,
+  fetchProfileSubscriptionDetails,
+} from "store/slices/profile.slice";
+
+import "./index.css";
+import { fetchProfileRunHistory, setImpersonate } from "store/slices/profile.slice";
+import HistoryContainer from "pages/testingHistory/components/HistoryContainer";
+import { clearRun } from "store/slices/testing.slice";
+
+const UserDetails = () => {
+  // const [data] = useState<any>(UserDetails);
+  const {
+    selectedUser: profile,
+    userSubscription,
+    loadingDetails,
+    loadingSubDetails,
+    detailsError,
+    updateSuccess,
+    loadingHistory,
+    runHistory,
+    impersonate
+  } = useAppSelector((state) => state.profile);
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+  const [open, setOpen] = useState(false);
+  const [title, setTitle] = useState("");
+  const [showToast, setShowToast] = useState(false);
+  const [currentSubscription, setCurrentSubscription] = useState<any>(null);
+
+  // Fetch user profile details if not already available
+  useEffect(() => {
+    const currentProfileId = window.location.href.split("/").pop()
```

Comment:
This can fail if for some reason there's a query string parameter on the url
Also the `react-router` lib has a method for accessing the url parameter in a more secure way
This can be done with the `useParams` hook from `react-router-dom`

Reply:
Like this `const { id } = useParams();`